cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/DiskFailureDetector.java

```diff
@@ -60,15 +60,15 @@ public DiskFailureDetector(Queue<Anomaly> anomalies, KafkaCruiseControl kafkaCru
    * @return The {@link AnomalyDetectionStatus anomaly detection status}, indicating whether the anomaly detector is ready.
    */
   private AnomalyDetectionStatus getDiskFailureDetectionStatus() {
-    int currentClusterGeneration = _kafkaCruiseControl.loadMonitor().clusterModelGeneration().clusterGeneration();
-    if (currentClusterGeneration == _lastCheckedClusterGeneration) {
+    ModelGeneration currentClusterModelGeneration = _kafkaCruiseControl.loadMonitor().clusterModelGeneration();
+    if (currentClusterModelGeneration.equals(_lastCheckedModelGeneration)) {
```

Comment:
In my opinion, we should check here if the `currentClusterModelGeneration` variable is null since at the first run the `_lastCheckedModelGeneration` will be also null so there could be a non required state. What do you think about that @lkalina-cera?

Another option is to add the following line to the constructor:
`_lastCheckedModelGeneration = new ModelGeneration(clusterAndGeneration.generation(), -1L);`

Reply:
I would like to use `_lastCheckedModelGeneration = new ModelGeneration(0, -1L)`. What Do you think?

Reply:
That sounds good to me!